sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -333,6 +333,8 @@ case class PrettyAttribute(
 }
 
 object VirtualColumn {
-  val groupingIdName: String = "grouping__id"
+  // The attribute name used by Hive, which has different result than Spark, deprecated.
+  val hiveGroupingIdName: String = "grouping__id"
+  val groupingIdName: String = "spark_grouping_id"
```

Comment:
"grouping__id" came from Hive, but unfortunately the implementation is wrong, see https://issues.apache.org/jira/browse/HIVE-12833. So we deprecated to favor the standard function grouping_id() as public API. "spark_grouping_id" is the virtual column only used internally.